apps/artemis-web/utils/token.tsx

```diff
@@ -303,3 +303,13 @@ export const GLOBAL_MEDIA_QUERIES = {
   pc: '(min-width: 700px)',
   mobile: '(max-width: 700px)',
 };
+
+export const autoLogout = () => {
+
+  setInterval(function () {
+    const timestamp = localStorage.getItem('login_timestamp');
+    const diff = 0.001 * (new Date().getTime() - new Date(timestamp).getTime());
+    if (diff > parseInt(process.env.NEXT_PUBLIC_SESSION_TIMEOUT, 10))
```

Comment:
I do not think this logic is correct. Essentially you enforce a hard timeout when the session timeout passes, but if the user does any manual action like visiting a page etc. this should be refreshed. Could you double-check how to implement this properly? The idea is that if there is no manual UI action that refreshes the session for a certain interval, then logout the user automatically. So if the user does sth in the UI, the counter should be somehow refreshed.

Reply:
You can check the following resources for examples on how to implement idle timeouts:

https://www.skptricks.com/2018/12/how-to-implement-auto-logout-client-side-in-react.html

https://javascript.plainenglish.io/building-an-auto-logout-session-timeout-with-react-using-hooks-e7804ef973ec

https://medium.com/tinyso/how-to-detect-inactive-user-to-auto-logout-by-using-idle-timeout-in-javascript-react-angular-and-b6279663acf2